### DISCUSSION TITLE
Do the Fedora upgrade docs need updating for users going to 41?

### DISCUSSION THREAD

**Discussion by khaytsus** [rpm-software-management/dnf5#1754] [General]
A user in the Fedora IRC channel has pointed out that in the [Fedora Upgrade docs](https://docs.fedoraproject.org/en-US/quick-docs/upgrading-fedora-offline/) there is at least one thing which no longer works in dnf5, dnf repoquery --unsatisfied, offhand don't know if there are other items on this page that don't work in dnf5, or the reason why this no longer works (deprecated, changed, etc) but it made me think that this page should be reviewed and potentially some edits made to make sure it's valid for both 39->40 (dnf4) and 40->41 (dnf5) users as this transition happens.

So not sure if dnf repoquery --unsatisfied is a bug or not, but either way, the upgrade docs should probably be reviewed to make sure users transitioning to 41 won't run into any issues.  Not even sure this is the right place to ask, but figure this may be the right team to review the docs and pass that information off to someone else if nothing else.

**Comment by Venemo** [comment]
Related bug report in the Fedora issue tracker: https://bugzilla.redhat.com/show_bug.cgi?id=2316290

**Comment by ppisar** [comment]
The best place is Fedora Bugzilla because it is a topic specific to Fedora.

The --unsatisfied option has no meaning for DNF because DNF does not allow to install a package with broken dependencies. That's probably the reason why nobody bother implementing it in DNF5. 

The only way how to get such broken package into a system is with "rpm --nodeps". Because the documentation is about upgrading a system and not about fixing a broken system, the best remedy is removing that option from the documentation.

I can review the documentation, but in general the documentation maybe challenging when it should apply to Fedoras with DNF4 and as well as with DNF5.

**Comment by Venemo** [comment]
> The only way how to get such broken package into a system is with "rpm --nodeps". Because the documentation is about upgrading a system and not about fixing a broken system, the best remedy is removing that option from the documentation.

I think the post-upgrade cleanup section in the article is meant to address any kind of potential breakage. If you feel that it's unnecessary to include the option to check for unsatisfied dependencies, that's a valid point. However the documentation should reflect what the software can actually do and the outdated recommendation should be removed.

(That said, I think there is value in including the option to fix a system that users broke by using `rpm --nodeps` but I realize that is off-topic here.)

> in general the documentation maybe challenging when it should apply to Fedoras with DNF4 and as well as with DNF5

I think it's OK to have a different chapter for upgrading from Fedora 40 to 41 vs. older releases. And the old parts can be removed after Fedora 40 is EOL.
